### PR TITLE
Improve docker support

### PR DESCRIPTION
**Pull request by johngian**

### CODE REVIEW

**Comment by johngian** [comment]
@akatsoulas @flamingspaz I refreshed a bit the docker support on remo

* Use scripts to run docker commands for dev/prod
* Use gunicorn for prod
* Limited the permissions to the docker user

File: docker-compose.yml [hunk 1]
```diff
@@ -15,7 +15,9 @@ services:
   broker:
     image: rabbitmq
   celery:
-    build: .
+    build:
+      context: .
+      dockerfile: docker/dev
     volumes:
       - .:/code
```
Comment (akatsoulas) [inline docker-compose.yml]: Let's move the celery command to an external file.

File: docker-compose.yml [hunk 2]
```diff
@@ -15,7 +15,9 @@ services:
   broker:
     image: rabbitmq
   celery:
-    build: .
+    build:
```
Comment (akatsoulas) [inline docker-compose.yml]: (nit) Should we use a celery image here?

Comment (johngian) [inline docker-compose.yml]: ``celery`` image is nothing more than a base python image with celery installed. In order to make it work you need to do complicated volume mounts to discover tasks. Also it doesn't support beat.

File: docker/dev [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
 RUN dpkg-reconfigure locales && locale-gen C.UTF-8 && \
     /usr/sbin/update-locale LANG=C.UTF-8
 RUN echo 'en_US.UTF-8 UTF-8' >> /etc/locale.gen && locale-gen
 ENV LC_ALL C.UTF-8
 ENV LANG en_US.UTF-8
 ENV LANGUAGE en_US.UTF-8
 
-COPY requirements /code/requirements/
-RUN pip install --require-hashes --no-deps -r requirements/dev.txt
-COPY . /code/
```
Comment (flamingspaz) [inline docker/dev]: Installing requirements and then copying the code was intentional to make the last layer really small. I think we should keep it like that so code changes only require a 5mb download.

Comment (johngian) [inline docker/dev]: Fixed a few lines below.

**Pull request merged**